SC_redesign/Crop_and_Soil/soil/nitrogen_cycling/leaching_runoff_erosion.py

```diff
@@ -0,0 +1,173 @@
+        -------
+        float
+            the concentration of the inorganic pools NO3/NH4 in the top soil layer (kg /mm H20)
+
+        References
+        -------
+        pseudocode_soil S.4.C.1
+        """
+        return soluble_nitrogen_amount * (1 - (exp(-soil_water_runoff_sum/saturation_content))/soil_water_runoff_sum)
+
+    @staticmethod
+    def _determine_nitrate_runoff_amount(nitrate_concentration: float,
+                                         runoff: float,
+                                         runoff_extraction_coef=0.1) -> float:
+        """This method determines the amount of nitrate runoff for the first layer
+
+        Parameters
+        ----------
+        nitrate_concentration: float
+            the concentration of the inorganic pools NO3/NH4 in the top soil layer (kg N/mm H20)
+        runoff_extraction_coef: float default for nitrate = 0.1
+            coefficient of extraction for runoff (unitless)
+        runoff: float
+            daily runoff of H2O (mm)
+
+        Returns
+        -------
+        float:
+            the amount of nitrate runoff from the first layer (kg/ha)
+
+        References
+        -------
+        pseudocode_soil S.4.C.2
+        """
+        return nitrate_concentration * runoff * runoff_extraction_coef
+
+    @staticmethod
+    def _determine_ammonium_runoff_amount(ammonium_concentration: float,
+                                          runoff: float,
+                                          runoff_extraction_coef=1) -> float:
+        """This method determines the amount of ammonium runoff for the first layer
+
+        Parameters
+        ----------
+        ammonium_concentration: float
+            the concentration of the inorganic pools NO3/NH4 in the top soil layer (kg N/mm H20)
+        runoff_extraction_coef: float default for NH4 = 1
+            coefficient of extraction for runoff (unitless)
+        runoff: float
+            daily runoff of H2O (mm)
+
+        Returns
+        -------
+        float:
+            the amount of ammonium runoff from the first layer (kg/ha)
+
+        References
+        -------
+        pseudocode_soil S.4.C.2
+        """
+        return ammonium_concentration * runoff * runoff_extraction_coef
+
+    @staticmethod
+    def _determine_nitrogen_erosion_concentration(nitrogen_amount: float,
+                                                  layer_thickness: float,
+                                                  bulk_density: float) -> float:
+        """This method calculates the soil nitrogen concentrations for the Fresh, Active, and Stable pools
+
+        Parameters
+        ----------
+        nitrogen_amount: float
+            amount of Fresh, Active, and Stable nitrogen (kg/ha)
+        bulk_density: float
+    `       bulk density of the soil layer (Mg per cubic meter)
+        Returns
+        -------
+        float
+            the soil nitrogen concentrations for the Fresh, Active, and Stable pools in soil(mg/kg)
```

Comment:
```suggestion
            the soil nitrogen concentrations for the Fresh, Active, and Stable pools in soil (g / metric ton soil)
```
I believe the pseudocode is wrong here, because it S.4.C.3 is exactly the same as 4:2.2.2, but SWAT specifies the units of conc_orgN as "(g N / metric ton soil)" in the paragraph beneath 4:2.2.1